rubicon/objc/runtime.py

```diff
@@ -1208,7 +1259,7 @@ def __new__(cls, object_ptr, _name=None, _bases=None, _ns=None):
             if is_block:
                 cls = ObjCBlockInstance
             else:
-                cls = cls._select_mixin(object_ptr)
+                cls = type_for_objcclass(libobjc.object_getClass(object_ptr))
```

Comment:
Is there any reason that the `is_block` check couldn't be merged into `type_for_objcclass`?

Reply:
Yes, unfortunately. `is_block` is assigned at the very top of `ObjCInstance.__new__`, and is used to remember whether the `object_ptr` that was originally passed to the constructor was of type `objc_block`. That information would be lost otherwise, because `object_ptr` is cast to `objc_id` in the next line.

Even if that wasn't the case, `type_for_objcclass` takes a *class* pointer, and the block check happens on the *instance* pointer. And the reason why we have to rely on the type of the instance pointer at all is because there's no good way to tell at runtime whether an object is a block or not.